dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -44,6 +44,17 @@ def color_for_status(status)
 Given /^I am on "([^"]*)"$/ do |url|
   url = replace_hostname(url)
   @browser.navigate.to "#{url}"
+  install_js_error_recorder
+end
+
+def install_js_error_recorder
+  @browser.execute_script(<<-JS
+  window.onerror = function (msg) {
+    window.detectedJSErrors = window.detectedJSErrors || [];
+    window.detectedJSErrors.push(msg);
+  };
```

Comment:
we might want to store the original window.onerror if it exists, and then also call it, so that if we depend on a different window.onerror defined elsewhere we're not blowing it away